crates/chia-consensus/src/gen/build_compressed_block.rs

```diff
@@ -0,0 +1,417 @@
+use crate::consensus_constants::ConsensusConstants;
+use chia_bls::Signature;
+use chia_protocol::SpendBundle;
+use clvmr::allocator::{Allocator, NodePtr};
+use clvmr::serde::{node_from_bytes_backrefs, Serializer};
+use std::io;
+
+#[cfg(feature = "py-bindings")]
+use pyo3::prelude::*;
+
+/// Maximum number of mempool items that can be skipped (not considered) during
+/// the creation of a block bundle. An item is skipped if it won't fit in the
+/// block we're trying to create.
+const MAX_SKIPPED_ITEMS: u32 = 6;
+
+/// Typical cost of a standard XCH spend. It's used as a heuristic to help
+/// determine how close to the block size limit we're willing to go.
+const MIN_COST_THRESHOLD: u64 = 6_000_000;
```

Comment:
Would it be reasonable to allow us to build the class with these parameters, and default to these values if they're not provided?

Reply:
yes, and ideally add whatever test you need to this repository as well, in rust. given that so much of the consensus logic lives in rust now this cannot be avoided. And I don't think it's an argument against moving more logic (and constants) into rust.

Reply:
I was thinking of two scenarios when I mentioned this:
1. Running a node to experiment with tweaking these values, where right now you're one local edit and a node restart away from achieving that.
2. End-to-end tests, AFAIK it's not currently possible to set those up in this repository, in rust.